businessRules/donations.js

```diff
@@ -2,14 +2,18 @@ const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
 const { getAllIdeas } = require('../models/ideas');
 const { getIdeas } = require('../utils/storage');
+const {
+    checkIfIdeaIdMatches,
+} = require('../validation/ideaFunctionValidation');
+const {
+    checkIfSumIsANumber,
+    checkIfIdeaIdMatchesIdea,
+} = require('../validation/donationFunctionValidation');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createDonationsWithDataValidation(firstName, sum, userId, ideaId) {
+    checkIfIdeaIdMatches(ideaId);
```

Comment:
The function name doesn't say what you're matching idea id to? Can you rename this function to make it clearer what it's doing?

Reply:
ideaThatMatchesIdeaId done. 
 Sorry, forgot to mark this one as commited.